trie/branchNode.go

```diff
@@ -336,11 +336,12 @@ func (bn *branchNode) commitCheckpoint(
 }
 
 func (bn *branchNode) commitSnapshot(
-	db common.DBWriteCacher,
+	db pruningStorer,
 	leavesChan chan core.KeyValueHolder,
 	ctx context.Context,
 	stats common.SnapshotStatisticsHandler,
 	idleProvider IdleNodeProvider,
+	saveToStorage bool,
```

Comment:
a bool in parameters list that conditions the execution path in this function. :man_facepalming:

Reply:
Refactored.